### PR TITLE
LORE-101 - allow internal requests to use WF set api

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/lore @mszabo-wikia 
https://wikia-inc.atlassian.net/browse/LORE-101

Updates `ApiWikiFactorySaveVariable` to allow internal requests without authentication to successfully save wf variables.

### CODE REVIEW

File: extensions/wikia/WikiFactory/api/ApiWikiFactorySaveVariable.php [hunk 1]
```diff
@@ -6,7 +6,7 @@ class ApiWikiFactorySaveVariable extends ApiBase {
 	 * @throws UsageException
 	 */
 	public function execute() {
-		if ( !$this->getUser()->isAllowed( 'wikifactory' ) ) {
+		if ( !$this->isInternalRequest() && !$this->getUser()->isAllowed( 'wikifactory' ) ) {
```
Comment (drsnyder) [inline extensions/wikia/WikiFactory/api/ApiWikiFactorySaveVariable.php:9]: Would this be more clear as `!($this->isInternalRequest() || $this->getUser()->isAllowed( 'wikifactory' ))`?

**Comment by drsnyder** [comment]
LGTM

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]
We should check `X-Mw-Auth-Ok` header too

File: extensions/wikia/WikiFactory/api/ApiWikiFactorySaveVariable.php (hunk 1 above)
Comment (nmonterroso) [reply extensions/wikia/WikiFactory/api/ApiWikiFactorySaveVariable.php:9]: I think that's a matter of personal preference. Personally, I think the less parens the easier it is to read code, especially here where I'd have to mentally expand the `!` to the two conditions.

**Comment by nmonterroso** [comment]
@mszabo-wikia can you explain what purpose `X-Mw-Auth-Ok` serves? I know we use it in several places to determine if a request is internal, but never really understood why.

**Comment by nmonterroso** [comment]
@mszabo-wikia I added the check for `X-Mw-Auth-Ok` header in 5780d9d2bdb

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**